projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java

```diff
@@ -43,25 +46,63 @@ public static void saveAgency(Path csvPath, String path) {
           isHeader = false;
           continue;
         }
-        Agency.Builder agency = Agency.newBuilder();
-        agency
-            .setAgencyId(Integer.parseInt(record.get("agency_id")))
-            .setAgencyName(record.get("agency_name"))
-            .setAgencyUrl(record.get("agency_url"))
-            .setAgencyTimezone(record.get("agency_timezone"))
-            .setAgencyLang(record.get("agency_lang"))
-            .setAgencyPhone(record.get("agency_phone"))
-            .setAgencyFareUrl(record.get("agency_fare_url"))
-            .build()
-            .writeDelimitedTo(output);
+        if (record.get("agency_id").equals(ISRAEL_RAIL_AGENCY_ID)) {
+          Agency.Builder agency = Agency.newBuilder();
+          agency
+              .setAgencyId(Integer.parseInt(record.get("agency_id")))
+              .setAgencyName(record.get("agency_name"))
+              .setAgencyUrl(record.get("agency_url"))
+              .setAgencyTimezone(record.get("agency_timezone"))
+              .setAgencyLang(record.get("agency_lang"))
+              .setAgencyPhone(record.get("agency_phone"))
+              .setAgencyFareUrl(record.get("agency_fare_url"))
+              .build()
+              .writeDelimitedTo(output);
+        }
       }
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
-  public static void saveCalendar(Path csvPath, String path) {
-    try (FileOutputStream output = new FileOutputStream(path)) {
+  private static String getDayOfToday() {
+    int day = Calendar.getInstance().get(Calendar.DAY_OF_WEEK);
+    switch (day) {
+      case Calendar.SUNDAY:
+        return "sunday";
+      case Calendar.MONDAY:
+        return "monday";
+      case Calendar.TUESDAY:
+        return "tuesday";
+      case Calendar.WEDNESDAY:
+        return "wednesday";
+      case Calendar.THURSDAY:
+        return "thursday";
+      case Calendar.FRIDAY:
+        return "friday";
+      case java.util.Calendar.SATURDAY:
+        return "saturday";
+      default:
+        return "";
+    }
+  }
+
+  private static boolean isServiceIdRelevantToday(CSVRecord record) {
+    DateFormat dateFormat = new SimpleDateFormat("yyyyMMdd");
+    Date date = new Date();
+    String todayString = getDayOfToday();
+    int startDate = Integer.parseInt(record.get("start_date"));
+    int endDate = Integer.parseInt(record.get("end_date"));
+    int currentDate = Integer.parseInt(dateFormat.format(date));
+    boolean isRecordDayTrueOnDayOfToday =
+        (Integer.parseInt(record.get(todayString)) == 1) ? true : false;
+    return (startDate <= currentDate
+        && currentDate <= endDate
+        && isRecordDayTrueOnDayOfToday == true);
```

Comment:
`isRecordDayTrueOnDayOfToday` variable is boolean. You don't need compare with `true`.